code/modules/antagonists/vampire/vampire_powers/dantalion_powers.dm

```diff
@@ -0,0 +1,198 @@
+
+/obj/effect/proc_holder/spell/vampire/thrall_commune/create_new_handler() //so thralls can use it
+	RETURN_TYPE(/datum/spell_handler)
+	return NONE
+
+/datum/spell_targeting/select_vampire_network/choose_targets(mob/user, obj/effect/proc_holder/spell/spell, params, atom/clicked_atom) // Returns the vampire and their thralls. If user is a thrall then it will look up their master's network
+	var/list/mob/living/targets = list()
+	var/datum/antagonist/vampire/V
+	V = user.mind.has_antag_datum(/datum/antagonist/vampire) // if the user is a vampire
+
+	if(!V)
+		for(var/datum/mind/M in user.mind.som.masters) // if the user is a thrall
+			V = M.has_antag_datum(/datum/antagonist/vampire)
+			if(V)
+				break
+
+	if(!V)
+		return
+
+	for(var/datum/mind/thrall in V.owner.som.serv)
+		targets += thrall.current
+	targets += V.owner.current
+	return targets
+
+/obj/effect/proc_holder/spell/vampire/thrall_commune/create_new_targeting()
+	var/datum/spell_targeting/select_vampire_network/T = new
+	return T
+
+/obj/effect/proc_holder/spell/vampire/thrall_commune/cast(list/targets, mob/user)
+	var/input = stripped_input(user, "Please choose a message to tell to the other thralls.", "Thrall Commune", "")
+	if(!input)
+		revert_cast(user)
+		return
+	var/message = "[user.real_name]: [input]"
+
+	for(var/mob/M in targets)
+		to_chat(M, "<span class='shadowling'>[message]</span>")
+	for(var/mob/M in GLOB.dead_mob_list)
+		to_chat(M, "<span class='shadowling'>([ghost_follow_link(user, ghost=M)]): [message] </span>")
+	log_say("(DANTALION) [input]", user)
+	user.create_log(SAY_LOG, "(DANTALION) [input]")
+
+/obj/effect/proc_holder/spell/vampire/pacify
+	name = "Pacify (10)"
+	desc = "Pacify a target temporarily, making them unable to cause harm."
+	gain_desc = "You have gained the ability to pacify someones harmful tendencies, preventing them from doing any physical harm to anyone."
+	charge_max = 30 SECONDS
+	required_blood = 10
+
+/obj/effect/proc_holder/spell/vampire/pacify/create_new_targeting()
+	var/datum/spell_targeting/click/T = new
+	T.range = 7
+	T.click_radius = 1
+	T.allowed_type = /mob/living/carbon/human
+	return T
+
+/obj/effect/proc_holder/spell/vampire/pacify/cast(list/targets, mob/user)
+	for(var/mob/living/carbon/human/H as anything in targets)
+		H.apply_status_effect(STATUS_EFFECT_PACIFIED)
+
+/obj/effect/proc_holder/spell/vampire/self/decoy
+	name = "Deploy Decoy (30)"
+	desc = "Briefly turn invisible and deploy a decoy illusion to fool your prey."
+	gain_desc = "You have gained the ability to turn invisible and create decoy illusions."
+	required_blood = 30
+	charge_max = 40 SECONDS
+
+/obj/effect/proc_holder/spell/vampire/self/decoy/cast(list/targets, mob/user)
+	var/mob/living/simple_animal/hostile/illusion/escape/E = new(get_turf(user))
+	E.Copy_Parent(user, 20, 20)
+	E.GiveTarget(user) //so it starts running right away
+	E.Goto(user, E.move_to_delay, E.minimum_distance)
```

Comment:
These two lines seem to always be used when making an escape illusion. Might be good to perhaps have them be integrated. Like if `GiveTarget` is called on `/mob/living/simple_animal/hostile/illusion/escape` then it also calls `Goto(...)`

Reply:
I am not sure I feel comfortable editing mob, especially AI code, like that. if a maintainer gives the thumbs up I will.